docs/design/baremetal/networking-infrastructure.md

```diff
@@ -70,7 +70,101 @@ DNS requirements internal to the cluster as possible, leaving only a
 small amount of public DNS configuration to be implemented by the user
 before starting the installation process.
 
-TODO - explain how this works ...
+In a `baremetal` environment, we do not know the IP addresses of all hosts in
+advance.  Those will come from an organization’s DHCP server.  Further, we can
+not rely on being able to program an organization’s DNS infrastructure in all
+cases.  We address these challenges in two ways:
 
-Explain that $cluster_name and $base_domain are supplied via
-install-config.yaml
+1. Self host some DNS infrastructure to provide DNS resolution for records only
+   needed internal to the cluster.
+2. Make use of mDNS (Multicast DNS) to dynamically discover the addresses of
+   hosts that we must resolve records for.
+
+### mdns-publisher
+
+https://github.com/openshift/mdns-publisher
+
+The `mdns-publisher` is the component that runs on each host to make itself
+discoverable by other hosts in the cluster.  It will be run by the
+`machine-config-operator`, once the following PR is complete:
+
+https://github.com/openshift/machine-config-operator/pull/795
+
+### coredns-mdns
+
+https://github.com/openshift/coredns-mdns/
+
+The `mdns` plugin for `coredns` was developed to perform DNS lookups
+based on discoverable information from mDNS.  This plugin will resolve both the
+`etcd-NNN` records, as well as the `_etcd-server-ssl._tcp.` SRV record.
```

Comment:
This begs the question of "what about api-int?" given the "internally resolvable" list above

Maybe move the api-int bit up to come before the mdns bit, especially since it's so simple?

Reply:
I mention the master-NN and worker-NN names above in mdns-publisher

So coredns-mdns publishes those too

Why? Should we include those in the list of "internally resolvable DNS records" above?

Reply:
I've moved `api-int` up as suggested, rewording it to not mention the "DNS VIP" since the DNS VIP will not have been discussed yet.

I intentionally left out mention of the `master-NN` and `worker-NN` names.  I know they were added because AWS had it, but I actually have no idea what they are used for, or if they're necessary at all.  Looking at the AWS implementation again, I don't think they are even there anymore.

@celebdor @cybertron It looks like we only need etcd records.  We may not need `mdns-publisher` on the workers at all.  Do you know of any reason that is still needed?

Reply:
I don't  know anything specific. The addresses we're serving are the ones that were in the original requirements.

I guess we can always try removing mdns-publisher from the workers and see what breaks.